lib/landingpage/productslider.dart

```diff
@@ -0,0 +1,175 @@
+import 'package:carousel_slider/carousel_slider.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_rating_bar/flutter_rating_bar.dart';
+
+import '../constants/constants.dart' as Constants;
+
+class ProductSlider extends StatefulWidget {
+  final productsList;
+
+  const ProductSlider({Key? key, this.productsList}) : super(key: key);
+
+  @override
+  State<ProductSlider> createState() => _MyWidgetState();
+}
+
+class _MyWidgetState extends State<ProductSlider> {
+  final buttonCarouselController = CarouselController();
+
+  static const buttonColor = 0xffe0e0e0;
+
+  @override
+  Widget build(BuildContext context) {
+    int productsCount = (widget.productsList.length / 4).round();
+    return Row(children: [
+      Padding(
+          padding: EdgeInsets.only(right: 15),
+          child: Container(
+              width: 35,
+              height: 35,
+              decoration: BoxDecoration(
+                borderRadius: BorderRadius.all(Radius.circular(100)),
+                color: Color(buttonColor)
+              ),
+              child: TextButton(
+              style: ButtonStyle(
+                shape: MaterialStateProperty.all<RoundedRectangleBorder>(
+                  RoundedRectangleBorder(
+                    borderRadius: BorderRadius.all(Radius.circular(100))
+                ))
+              ),
+                onPressed: () => buttonCarouselController.nextPage(
+                    duration: Duration(milliseconds: 300),
+                    curve: Curves.linear),
+                child: Text(
+                  '←',
+                  style: TextStyle(color: Colors.black),
+                  textAlign: TextAlign.center,
+                ),
+              ))),
+      Expanded(
+        child: CarouselSlider.builder(
+          carouselController: buttonCarouselController,
+          options: CarouselOptions(
+              aspectRatio: 2.0, enlargeCenterPage: false, viewportFraction: 1),
+          itemCount: productsCount,
+          itemBuilder: (context, index, realIndex) {
+            final int first = index;
+            final int second = first + 1;
+            return Row(
+              children: [for (var i = 4 * first; i <= 4 * second - 1; i += 1) i]
```

Comment:
I'm not quite sure how this is working, but it doesn't seem to be generic to me. All the data about "distribuitor", "3 review-uri" and so on, would be data that will come from the backend. Try preparing this by making the component accept some sort of parameters and extract the hard coded values outside. Pretty sure that's the approach you used above.

Reply:
It's generic, but it's designed to show 4 products at a time

Reply:
*maximum, and can work with many more